plugins/deployment-helm-chart/src/index.ts

```diff
@@ -0,0 +1,137 @@
+  DsgContext,
+  Events,
+  Module,
+} from "@amplication/code-gen-types";
+import {
+  applicationVersionKey,
+  chartVersionKey,
+  configurationKey,
+  hostKey,
+  portKey,
+  repositoryKey,
+  serviceNameKey,
+  tagKey,
+} from "./constants";
+import { join } from "node:path";
+import { getPluginSettings } from "./utils";
+import { resolve } from "path";
+import { EventNames } from "@amplication/code-gen-types";
+
+class HelmChartPlugin implements AmplicationPlugin {
+  register(): Events {
+    return {
+      [EventNames.CreateServerDotEnv]: {
+        after: this.afterCreateServerDotEnv,
+      },
+      // [EventNames.CreateAdminUIDotEnv]: {
+      //   after: this.afterCreateAdminUIDotEnv,
+      // },
+    };
+  }
+
+  async afterCreateServerDotEnv(
+    context: DsgContext,
+    eventParams: CreateServerDotEnvParams,
+    modules: Module[]
+  ) {
+    context.logger.info(`Generating Helm Chart...`);
+
+    // determine the name of the service which will be used as the name for the helm chart
+    // chart names must be lower case letters and numbers. words may be separated with dashes (-):
+    const serviceName = context.resourceInfo?.name
+      .toLocaleLowerCase()
+      .replaceAll(/[^a-zA-Z0-9-]/g, "-");
+
+    if (!serviceName) {
+      throw new Error("Service name is undefined");
+    }
+
+    // fetch user settings + merge with default settings
+    const settings = getPluginSettings(context.pluginInstallations);
+
+    // fetch the variables from the .env file and create a variable for it which can be
+    // used to fill the configmap, secrets should be moved by the users themself
+    const variables = eventParams.envVariables;
+
+    if (!variables) {
+      throw new Error("Variables can't be fetched");
+    }
+
+    const configmapIndentation = "    ";
+    let configmap: string = "";
+
+    variables.forEach((variable) => {
+      const [name, value] = Object.entries(variable)[0];
+      configmap = `${configmap}\n${configmapIndentation}${name}: ${value}`;
+    });
+
+    /**
+     * save the renderedOutput to the desired directory the options are on the root of the repository
+     * and within the directory of the services itself setting "root_directory":
+     *    option 1 (value: true):  /<directory_name_value>/<service_name>
+     *         create directory on the root of the repository (../) with the name provided
+     *         through the settings.directory_name, subsequently place all of the files that
+     *         from the static directory via the renderdOutput variable
+     *    option 2 (value: false): /<service_name>/<directory_name_value>/<service_name>
+     *         create directory within the directory of the service with the name provided
+     *         through the settings.directory_name, subsequently place all of the files that
+     *         from the static directory via the renderdOutput variable
+     */
+    let helmDirectoryPath = "";
+    if (settings.root_level == true) {
```

Comment:
While root_level is actually defined as Boolean in the interface, isn't it better to always use === for extra safety? (same is true for line 90)

Reply:
Didn't acutally know about this concept, thanks for the pointer.  I enforced this.